authn/saml/src/main/java/se/swedenconnect/signservice/authn/saml/SwedenConnectSamlAuthenticationHandler.java

```diff
@@ -104,6 +118,22 @@ protected AuthnRequestGeneratorContext createAuthnRequestContext(
     //
     final PrincipalSelection principalSelection = this.getPrincipalSelection(authnRequirements, context, idpMetadata);
 
+    // Should we send a SAD request?
+    //
+    final SADRequest sadRequest = authnRequirements.getSignatureActivationRequestData() != null
+        && this.isSignatureActivationProtocolSupported(idpMetadata)
+            ? (SADRequest) XMLObjectSupport.buildXMLObject(SADRequest.DEFAULT_ELEMENT_NAME)
+            : null;
+    if (sadRequest != null) {
+      sadRequest.setRequesterID(this.authnRequestGenerator.getSpEntityID());
+      sadRequest.setSignRequestID(authnRequirements.getSignatureActivationRequestData().getSignRequestId());
+      sadRequest.setDocCount(1);
+      sadRequest.setID(UUID.randomUUID().toString());
```

Comment:
This must be wrong. docCount should not be always 1 but reflect the actual number of documents being signed